integration/kotlinx-coroutines-jdk8/src/test/kotlin/kotlinx/coroutines/experimental/future/FutureTest.kt

```diff
@@ -255,6 +255,49 @@ class FutureTest : TestBase() {
         assertThat(future.get(), IsEqual("OK"))
     }
 
+    @Test
+    fun testCompletableFutureStageAsDeferred() = runBlocking {
+        val deferred: Deferred<Int> = CompletableFuture.supplyAsync { Thread.sleep(100) ; 42 }.asDeferred()
```

Comment:
A Lock is more reliable than a `sleep`.
You can
1. use a Lock to pause the Supplier
2. check Deferred as incomplete
3. release the Lock to resume the Supplier
4. verify Deferred result